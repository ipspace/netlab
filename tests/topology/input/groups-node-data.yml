defaults:
  device: cumulus
  
module: [ bgp ]

defaults.bgp.as: 65000

nodes: [ a,b,c,d,e,f ]

groups:
  g1: [ a,b,c ]
  g2:
    members: [ d,e,f ]
    node_data:
      bgp.as: 65001
