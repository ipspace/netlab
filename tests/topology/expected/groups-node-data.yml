bgp:
  advertise_loopback: true
  as: 65000
  community:
    ebgp:
    - standard
    ibgp:
    - standard
    - extended
  next_hop_self: true
groups:
  g1:
    members:
    - a
    - b
    - c
  g2:
    members:
    - d
    - e
    - f
    node_data:
      bgp:
        as: 65001
input:
- topology/input/groups-node-data.yml
- package:topology-defaults.yml
module:
- bgp
name: input
nodes:
  a:
    bgp:
      advertise_loopback: true
      as: 65000
      community:
        ebgp:
        - standard
        ibgp:
        - standard
        - extended
      ipv4: true
      neighbors:
      - as: 65000
        ipv4: 10.0.0.2
        name: b
        type: ibgp
      - as: 65000
        ipv4: 10.0.0.3
        name: c
        type: ibgp
      next_hop_self: true
      router_id: 10.0.0.1
    box: CumulusCommunity/cumulus-vx
    device: cumulus
    id: 1
    loopback:
      ipv4: 10.0.0.1/32
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.101
      mac: 08-4F-A9-00-00-01
    module:
    - bgp
    name: a
  b:
    bgp:
      advertise_loopback: true
      as: 65000
      community:
        ebgp:
        - standard
        ibgp:
        - standard
        - extended
      ipv4: true
      neighbors:
      - as: 65000
        ipv4: 10.0.0.1
        name: a
        type: ibgp
      - as: 65000
        ipv4: 10.0.0.3
        name: c
        type: ibgp
      next_hop_self: true
      router_id: 10.0.0.2
    box: CumulusCommunity/cumulus-vx
    device: cumulus
    id: 2
    loopback:
      ipv4: 10.0.0.2/32
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.102
      mac: 08-4F-A9-00-00-02
    module:
    - bgp
    name: b
  c:
    bgp:
      advertise_loopback: true
      as: 65000
      community:
        ebgp:
        - standard
        ibgp:
        - standard
        - extended
      ipv4: true
      neighbors:
      - as: 65000
        ipv4: 10.0.0.1
        name: a
        type: ibgp
      - as: 65000
        ipv4: 10.0.0.2
        name: b
        type: ibgp
      next_hop_self: true
      router_id: 10.0.0.3
    box: CumulusCommunity/cumulus-vx
    device: cumulus
    id: 3
    loopback:
      ipv4: 10.0.0.3/32
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.103
      mac: 08-4F-A9-00-00-03
    module:
    - bgp
    name: c
  d:
    bgp:
      advertise_loopback: true
      as: 65001
      community:
        ebgp:
        - standard
        ibgp:
        - standard
        - extended
      ipv4: true
      neighbors:
      - as: 65001
        ipv4: 10.0.0.5
        name: e
        type: ibgp
      - as: 65001
        ipv4: 10.0.0.6
        name: f
        type: ibgp
      next_hop_self: true
      router_id: 10.0.0.4
    box: CumulusCommunity/cumulus-vx
    device: cumulus
    id: 4
    loopback:
      ipv4: 10.0.0.4/32
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.104
      mac: 08-4F-A9-00-00-04
    module:
    - bgp
    name: d
  e:
    bgp:
      advertise_loopback: true
      as: 65001
      community:
        ebgp:
        - standard
        ibgp:
        - standard
        - extended
      ipv4: true
      neighbors:
      - as: 65001
        ipv4: 10.0.0.4
        name: d
        type: ibgp
      - as: 65001
        ipv4: 10.0.0.6
        name: f
        type: ibgp
      next_hop_self: true
      router_id: 10.0.0.5
    box: CumulusCommunity/cumulus-vx
    device: cumulus
    id: 5
    loopback:
      ipv4: 10.0.0.5/32
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.105
      mac: 08-4F-A9-00-00-05
    module:
    - bgp
    name: e
  f:
    bgp:
      advertise_loopback: true
      as: 65001
      community:
        ebgp:
        - standard
        ibgp:
        - standard
        - extended
      ipv4: true
      neighbors:
      - as: 65001
        ipv4: 10.0.0.4
        name: d
        type: ibgp
      - as: 65001
        ipv4: 10.0.0.5
        name: e
        type: ibgp
      next_hop_self: true
      router_id: 10.0.0.6
    box: CumulusCommunity/cumulus-vx
    device: cumulus
    id: 6
    loopback:
      ipv4: 10.0.0.6/32
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.106
      mac: 08-4F-A9-00-00-06
    module:
    - bgp
    name: f
provider: libvirt
